### ISSUE TITLE
HTTPFile._fetch_range raises when passed headers kwarg

### ISSUE THREAD

**Issue by rpetchler** [dask/dask#4479]
Reproducible example:

```python
from dask.bytes.http import HTTPFile

url = "http://www.example.com/"
headers = {"DNT": "1"}

fobj = HTTPFile(url=url, headers=headers)
print(fobj._fetch_range(0, 64))
```

Raised exception:

```
Traceback (most recent call last):
  File "main.py", line 7, in <module>
    print(fobj._fetch_range(0, 64))
  File "dask/bytes/http.py", line 249, in _fetch_range
    r = self.session.get(self.url, headers=headers, stream=True, **kwargs)
TypeError: get() got multiple values for keyword argument 'headers'
```

The cause appears to be a typo in [these lines](https://github.com/dask/dask/blob/1.1.1/dask/bytes/http.py#L246-L249). The correct implementation calls `kwargs.pop`, not `self.kwargs.pop`.

**Comment by martindurant** [comment]
`HTTPFile` isn't really meant to be used except for internal Dask access to remote data. You may well be right nevertheless, but I'd still like to know how you came across this class and what your intended use is.

(Would [fsspec's version](https://github.com/martindurant/filesystem_spec/blob/master/fsspec/implementations/http.py) be useful to you?)

**Comment by rpetchler** [comment]
Thanks for your prompt reply, @martindurant.

Here's a larger example that illustrates how passing headers through `storage_options` raises this exception:

```python
import dask
import dask.bag as db

dask.config.set(scheduler='synchronous')

urls = [
    "http://www.example.com/",
    "http://www.example.com/",
]

headers = {"DNT": "1"}
storage_options = {"headers": headers}

bag = db.read_text(urls, storage_options=storage_options)
bag.map(len).sum().compute()
```

And here's the corresponding trace:

```
Traceback (most recent call last):
  File "httpfile.py", line 15, in <module>
    bag.map(len).sum().compute()
  File "dask/base.py", line 156, in compute
    (result,) = compute(self, traverse=False, **kwargs)
  File "dask/base.py", line 398, in compute
    results = schedule(dsk, keys, **kwargs)
  File "dask/local.py", line 501, in get_sync
    return get_async(apply_sync, 1, dsk, keys, **kwargs)
  File "dask/local.py", line 447, in get_async
    fire_task()
  File "dask/local.py", line 443, in fire_task
    callback=queue.put)
  File "dask/local.py", line 490, in apply_sync
    res = func(*args, **kwds)
  File "dask/local.py", line 235, in execute_task
    result = pack_exception(e, dumps)
  File "dask/local.py", line 230, in execute_task
    result = _execute_task(task, data)
  File "dask/core.py", line 119, in _execute_task
    return func(*args2)
  File "dask/bag/core.py", line 2118, in empty_safe_apply
    _, part = peek(part)
  File "venv/lib/python3.7/site-packages/toolz/itertoolz.py", line 944, in peek
    item = next(iterator)
  File "dask/bag/core.py", line 1745, in map_chunk
    for a in zip(*args):
  File "dask/bag/text.py", line 105, in file_to_blocks
    for line in f:
  File "dask/bytes/http.py", line 185, in read
    self. _fetch(self.loc, end)
  File "dask/bytes/http.py", line 196, in _fetch
    self.cache = self._fetch_range(start, self.end)
  File "dask/bytes/http.py", line 249, in _fetch_range
    r = self.session.get(self.url, headers=headers, stream=True, **kwargs)
TypeError: get() got multiple values for keyword argument 'headers'
```

I encountered this error while working with a proprietary S3-like object store that uses headers to authenticate requests.

**Comment by martindurant** [comment]
OK, got you. You are quite right. Would you like to create a PR?

**Comment by rpetchler** [comment]
Closed in #4480.
